test/e2e/framework/framework.go

```diff
@@ -376,6 +376,21 @@ func (f *Framework) AfterEach() {
 		}
 	}
 
+	if TestContext.GatherApiserverProfilesAfterTest {
+		By("Gathering CPU and memory allocation profiles of the apiserver")
+		if TestContext.ReportDir == "" {
+			Logf("Skipping apiserver profile gathering as no ReportDir path has been specified")
+		} else {
+			// Gather the various profiles from master and place them under ReportDir.
+			if err := GatherApiserverCPUProfile(f.BaseName); err != nil {
```

Comment:
CPU profile read after the test was completed might not be the most interesting thing in the world. But I'm fine with adding that if you think it's helpful.

Reply:
+1 to Marek`s comment. No real usecase for it comes to my mind now.

Reply:
Good catch - I actually just realized that the profile is over the last ~30s of running. What I can do is to change it to not be called as part of AfterEach(), but still leave the utility function up there, so that someone can call it from within their test.

SG?

Reply:
I don't think a profile of an idle cluster is useful. What I *would* find useful would be a library which would:
1. allow me to insert (manually) a single `GatherApiserverCPUProfile(N * time.Second)` call anywhere inside a test where I think would be useful, and
2. make sure it is automatically transformed into a PDF graph stored neatly with other artifacts at the end of the test
without thinking about details other than the duration of the profile

Reply:
@porridge SGTM. Having just a library there was more or less what I was suggesting above. I can change it to directly add the PDF (though I initially thought that having the raw data would be more informative, but I guess that'd be less convenient).

Reply:
Made the change.